baseframe/filters.py

```diff
@@ -166,3 +168,25 @@ def cdata(text):
     Convert text to a CDATA sequence
     """
     return Markup('<![CDATA[' + text.replace(']]>', ']]]]><![CDATA[>') + ']]>')
+
+
+@baseframe.app_template_filter('shortdate')
+def shortdate(value):
+    if isinstance(value, datetime):
+        dt = utc.localize(value).astimezone(get_timezone())
+        utc_now = datetime.now(utc)
+    else:
+        dt = value
+        utc_now = request_timestamp().date()
+    if dt > (utc_now - timedelta(days=int(current_app.config.get('SHORTDATE_THRESHOLD_DAYS', 0)))):
+        return dt.strftime('%e %b')
+    else:
+        # The string replace hack is to deal with inconsistencies in the underlying
+        # implementation of strftime. See https://bugs.python.org/issue8304
+        return six.text_type(dt.strftime("%e %b '%y")).replace(u"'", u"’")
```

Comment:
The use of an apostrophe is likely to hurt localization in languages that don't use an apostrophe to represent abbreviations.